vscode/src/ruby/versionManager.ts

```diff
@@ -82,30 +113,98 @@ export abstract class VersionManager {
 
   // Runs the given command in the directory for the Bundle, using the user's preferred shell and inheriting the current
   // process environment
-  protected runScript(command: string) {
-    let shell: string | undefined;
-
-    // If the user has configured a default shell, we use that one since they are probably sourcing their version
-    // manager scripts in that shell's configuration files. On Windows, we never set the shell no matter what to ensure
-    // that activation runs on `cmd.exe` and not PowerShell, which avoids complex quoting and escaping issues.
-    if (vscode.env.shell.length > 0 && os.platform() !== "win32") {
-      shell = vscode.env.shell;
-    }
+  protected runScript(command: string, options: ExecOptions = {}) {
+    const execOptions = this.execOptions(options);
 
     this.outputChannel.info(
-      `Running command: \`${command}\` in ${this.bundleUri.fsPath} using shell: ${shell}`,
+      `Running command: \`${command}\` in ${execOptions.cwd} using shell: ${execOptions.shell}`,
     );
     this.outputChannel.debug(
-      `Environment used for command: ${JSON.stringify(process.env)}`,
+      `Environment used for command: ${JSON.stringify(execOptions.env)}`,
     );
 
-    return asyncExec(command, {
-      cwd: this.bundleUri.fsPath,
-      shell,
-      env: process.env,
+    return asyncExec(command, execOptions);
+  }
+
+  protected runRubyScript(
+    rubyCommand: string,
+    script: string,
+  ): Promise<{ stdout: string; stderr: string }> {
+    return new Promise((resolve, _reject) => {
+      this.outputChannel.info(
+        `Ruby \`${rubyCommand}\` running Ruby script: \`${script}\``,
+      );
+
+      const { command, args, env } = this.parseCommand(rubyCommand);
+      const ruby = spawn(command, args, this.execOptions({ env }));
+
+      ruby.stdout.once("data", (data) => {
+        resolve({ stdout: data.toString(), stderr: "" });
+      });
+      ruby.stderr.once("data", (data) => {
+        resolve({ stdout: "", stderr: data.toString() });
+      });
```

Comment:
these are streams, they may emit more than one event
https://nodejs.org/docs/latest/api/child_process.html#subprocessstderr

if you want to capture all the output, you probably need to read them all to completion